### PR TITLE
Add TEI Images

### PR DESCRIPTION
**Pull request by philschmid**
# What does this PR do?

This PR adds the first TEI-supported images to the official SageMaker DLCs. The Images can be built similarly and run similar to TGI.

### CODE REVIEW

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 7fb7a8f8b7f9ba6c241f9cf74ff18b8f72764eac
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=054f79b9-ac80-4964-8f31-a2e5fd0f5b2d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 3ee16c03a3cffb4e23b5abe58b9519dec6dd8b50
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d1a2f783-c088-49bd-af8b-ab3459b69ed4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 0bcfcf7595312bfa0e3c56530c7b7c9717d0f639
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=40ed3e51-fa24-477f-8de8-afa340569232%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 6bf5b4f40e28116568bee2bf0e42d70f60fd355b
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=3d0916c1-e480-4904-b9c9-46dfe022967b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+FROM base-builder AS builder
+
+ARG GIT_SHA
+ARG DOCKER_LABEL
+ARG VERTEX
+
+# sccache specific variables
+ARG ACTIONS_CACHE_URL
+ARG ACTIONS_RUNTIME_TOKEN
+ARG SCCACHE_GHA_ENABLED
```
Comment (OlivierDehaene) [inline huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu:41]:
```suggestion
ARG SCCACHE_GHA_ENABLED

# limit the number of kernels built at the same time
ARG RAYON_NUM_THREADS=2

```

File: huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+WORKDIR /usr/src
+
+COPY --from=planner /usr/src/recipe.json recipe.json
+
+FROM builder as builder-75
+
+RUN CUDA_COMPUTE_CAP=75 cargo chef cook --release --features candle-cuda-turing --no-default-features --recipe-path recipe.json && sccache -s
```
Comment (OlivierDehaene) [inline huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu:49]:
```suggestion
RUN CUDA_COMPUTE_CAP=75 cargo chef cook --release --features candle-cuda-turing --features http --no-default-features --recipe-path recipe.json && sccache -s
```

File: huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+COPY core core
+COPY router router
+COPY Cargo.toml ./
+COPY Cargo.lock ./
+
+RUN CUDA_COMPUTE_CAP=75 cargo build --release --bin text-embeddings-router -F candle-cuda-turing -F http --no-default-features && sccache -s
+
+FROM builder as builder-80
+
+RUN CUDA_COMPUTE_CAP=80 cargo chef cook --release --features candle-cuda-turing --no-default-features --recipe-path recipe.json && sccache -s
```
Comment (OlivierDehaene) [inline huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu:61]:
```suggestion
RUN CUDA_COMPUTE_CAP=80 cargo chef cook --release --features candle-cuda --features http --no-default-features --recipe-path recipe.json && sccache -s
```

File: huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+COPY backends backends
+COPY core core
+COPY router router
+COPY Cargo.toml ./
+COPY Cargo.lock ./
+
+RUN CUDA_COMPUTE_CAP=80 cargo build --release --bin text-embeddings-router -F candle-cuda-turing -F http --no-default-features && sccache -s
```
Comment (OlivierDehaene) [inline huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu:69]:
```suggestion
RUN CUDA_COMPUTE_CAP=80 cargo build --release --bin text-embeddings-router -F candle-cuda -F http --no-default-features && sccache -s
```

File: huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+FROM builder as builder-90
+
+RUN CUDA_COMPUTE_CAP=90 cargo chef cook --release --features candle-cuda-turing --no-default-features --recipe-path recipe.json && sccache -s
```
Comment (OlivierDehaene) [inline huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu:73]:
```suggestion
RUN CUDA_COMPUTE_CAP=90 cargo chef cook --release --features candle-cuda --features http --no-default-features --recipe-path recipe.json && sccache -s
```

File: huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+COPY backends backends
+COPY core core
+COPY router router
+COPY Cargo.toml ./
+COPY Cargo.lock ./
+
+RUN CUDA_COMPUTE_CAP=90 cargo build --release --bin text-embeddings-router -F candle-cuda-turing -F http --no-default-features && sccache -s
```
Comment (OlivierDehaene) [inline huggingface/pytorch/tgi/tei/docker/1.2.1/cu122/Dockerfile.gpu:81]:
```suggestion
RUN CUDA_COMPUTE_CAP=90 cargo build --release --bin text-embeddings-router -F candle-cuda -F http --no-default-features && sccache -s
```

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: b2533798a4287f4864ff2d6a2000a6ffe9a714f3
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=e48a4213-c80d-476d-b56b-14f86da04ad0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: f62ba5a088964ae73d410daedee515f5ae4ae865
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=aa79c712-5f97-435d-b5f8-d4d9b129c6fa%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 8756527a08d33b9c90e0d678487918f14a72b9db
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2c2ea234-fcca-4f65-9242-1aca409a5b51%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 80442692bf5db06b1e23727932be0dbe05f0754d
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8f9cff5c-7bf5-47ed-a841-d3eef5edc0ea%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: cfb7b89e0f8fa56695e72fa31f9949ac945c9343
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=cd31433f-d3e9-4340-955c-5f07a70b7e33%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: c39871a1bc597fb1fa7c7e0d15b0c608f799f8de
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8b67013f-26fe-4eef-a3f9-8a31464069ba%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 1bdb2345a97611cadede4a9a1644e960170cf3d4
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c53eacb8-1641-4387-858f-09481a203f63%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by amzn-choeric** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: tgi-pr-inf2
* Commit ID: 02386ba2c22bb57d46a14f492604f3caf239520e
* Result: FAILED
* [Build Logs](https://cua665myu4.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8397443b-95c5-45d5-b85b-bb132aff80f5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request closed without merging**